test/suites/integration/subnet_test.go

```diff
@@ -54,6 +54,36 @@ var _ = Describe("Subnets", func() {
 
 		env.ExpectInstance(pod.Spec.NodeName).To(HaveField("SubnetId", HaveValue(Equal(firstSubnet))))
 	})
+	It("should use the subnet-id selector with resource based naming", func() {
+		subnets := getSubnets(map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName})
+		Expect(len(subnets)).ToNot(Equal(0))
+		availabilityZone := lo.Keys(subnets)
+
+		var allSubnets []string
+		for zoneIndex := 0; zoneIndex < len(availabilityZone); zoneIndex++ {
+			allSubnets = append(allSubnets, subnets[availabilityZone[zoneIndex]]...)
+		}
```

Comment:
I think we can remove this based on the comment above, but in the future, you can use a `range` which is a lot more common in go for-loops. 

```
for _, zone := range availabilityZone {
    allSubnets = append(allSubnets, subnets[zone]...)
} 
```

or even better, `range` supports maps as well:

```
for zone, subnetIDs := range subnets {
    allSubnets = append(allSubnets, subnetIDs...)
}
```

Here's some more examples: https://gobyexample.com/range